app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java

```diff
@@ -227,6 +226,18 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
             updateUndoRedoIconStates();
         });
         _hlEditor.addTextChangedListener(GsTextWatcherAdapter.after(s -> debounced.run()));
+
+        // Restore keyboard when we regain focus
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+            _hlEditor.getViewTreeObserver().addOnWindowFocusChangeListener(hasFocus -> {
```

Comment:
Much cleaner way to restore keyboard state. Removed older method from like 10 places.